xdsl/interpreters/experimental/wgsl_printer.py

```diff
@@ -46,9 +47,12 @@ def _(self, op: gpu.FuncOp, out_stream: IO[str]):
                 arg_type = "f32"
             elif arg.type == builtin.IndexType():
                 arg_type = "u32"
-            elif isinstance(arg.type, MemRefType):
-                memref_typ = cast(MemRefType[Attribute], arg.type)
-                arg_type = f"array<{memref_typ.element_type}>"
+            elif isa(arg.type, MemRefType[Attribute]):
```

Comment:
```suggestion
            elif isinstance(arg.type, MemrefType):
```
It looks like the type of the generic parameter is not really tested against, so might as well use the simpler `isinstance`? This is a nit